pkg/services/alerting/eval_handler.go

```diff
@@ -60,6 +61,40 @@ func (e *DefaultEvalHandler) Eval(context *EvalContext) {
 	context.Firing = firing
 	context.NoDataFound = noDataFound
 	context.EndTime = time.Now()
+	context.Rule.State = e.overrideState(context)
```

Comment:
not sure I understand the naming. Reading this as your overriding some behavior or default state? why not e.getNewCurrentState() or determineNewState()